cluster/defaults.go

```diff
@@ -273,6 +276,32 @@ func (c *Cluster) setClusterImageDefaults() error {
 	return nil
 }
 
+func (c *Cluster) setClusterDNSDefaults() error {
+	if c.DNS == nil || len(c.DNS.Provider) == 0 {
+		clusterSemVer, err := util.StrToSemVer(c.Version)
+		if err != nil {
+			return err
+		}
+		logrus.Debugf("No DNS provider configured, setting default based on cluster version [%s]", clusterSemVer)
+		K8sVersionCoreDNSSemVer, err := util.StrToSemVer(K8sVersionCoreDNS)
+		if err != nil {
+			return err
+		}
+		// Default DNS provider for cluster version 1.14.0 and higher is coredns
+		ClusterDNSProvider := CoreDNSProvider
+		// If cluster version is less than 1.14.0 (K8sVersionCoreDNSSemVer), use kube-dns
+		if clusterSemVer.LessThan(*K8sVersionCoreDNSSemVer) {
```

Comment:
also might be better reversed, use the Default and override it at the condition.

Reply:
Due to a vendoring issue, we have an older `go-semver` lib with any `LessThan` available. I can reverse it by using `! if clusterSemVer.LessThan` but that seemed more ugly than this. If you still rather have that I can change it.

The vendoring issue will take more time to investigate and fix appropriately (and backport)

Reply:
Not worth it. It was just a nit.